jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -109,29 +128,64 @@ private static Properties loadPropertiesFromPath(String path) throws Configurati
     }
   }
 
-  JmxScraper(JmxConnectorBuilder client) {
+  JmxScraper(JmxConnectorBuilder client, JmxMetricInsight service, JmxScraperConfig config) {
     this.client = client;
+    this.service = service;
+    this.config = config;
   }
 
   private void start() throws IOException {
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread(
+                () -> {
+                  logger.info("JMX scraping stopped");
+                  running.set(false);
+                }));
+
+    try (JMXConnector connector = client.build()) {
+      MBeanServerConnection connection = connector.getMBeanServerConnection();
+      service.startRemote(getMetricConfig(config), () -> Collections.singletonList(connection));
+
+      running.set(true);
+      logger.info("JMX scraping started");
+
+      while (running.get()) {
+        try {
+          Thread.sleep(100);
+        } catch (InterruptedException e) {
+          // silently ignored
+        }
+      }
+    }
+  }
 
-    JMXConnector connector = client.build();
-
-    @SuppressWarnings("unused")
-    MBeanServerConnection connection = connector.getMBeanServerConnection();
-
-    // TODO: depend on instrumentation 2.9.0 snapshot
-    // MetricConfiguration metricConfig = new MetricConfiguration();
-    // TODO create JMX insight config from scraper config
-    // service.startRemote(metricConfig, () -> Collections.singletonList(connection));
+  private static MetricConfiguration getMetricConfig(JmxScraperConfig scraperConfig) {
+    MetricConfiguration config = new MetricConfiguration();
+    for (String system : scraperConfig.getTargetSystems()) {
+      try {
+        addRulesForSystem(system, config);
+      } catch (RuntimeException e) {
+        logger.warning("unable to load rules for system " + system + ": " + e.getMessage());
+      }
+    }
+    // TODO : add ability for user to provide custom yaml configurations
 
-    logger.info("JMX scraping started");
+    return config;
+  }
 
-    // TODO: wait a bit to keep the JVM running, this won't be needed once calling jmx insight
-    try {
-      Thread.sleep(5000);
-    } catch (InterruptedException e) {
-      throw new IllegalStateException(e);
+  private static void addRulesForSystem(String system, MetricConfiguration conf) {
+    String yamlResource = system + ".yaml";
+    try (InputStream inputStream =
+        JmxScraper.class.getClassLoader().getResourceAsStream(yamlResource)) {
+      if (inputStream != null) {
+        RuleParser parserInstance = RuleParser.get();
+        parserInstance.addMetricDefsTo(conf, inputStream, system);
+      } else {
+        throw new IllegalStateException("no support for " + system);
```

Comment:
I think IllegalStateException was designed for something else. I think ConfigurationException would be better (or at least IllegalArgumentException).